### PR TITLE
feat: Geocoder autocompletion

### PR DESCRIPTION
**Pull request by aanev**
Add geocodeAutocomplete and findLocationById methods to HereMapsAPIService public API.

### CODE REVIEW

**Comment by dverbovyi** [comment]
@aanev thanks for your pull request

Your API changes would not be spread via map-api instance, because you need to add this API via MapProxy like here
https://github.com/dverbovyi/angular-heremaps/blob/master/src/heremaps.directive.js#L165

File: src/providers/api.service.js [hunk 1]
```diff
@@ -123,6 +125,66 @@ function HereMapsAPIService($q, $http, HereMapsConfig, HereMapsUtilsService, Her
         return deferred.promise;
     }
 
+    function geocodeAutocomplete(params) {
+        if (!params)
+            return console.error('Missing required parameters');
+
+        var autocompleteUrl = protocol + '://autocomplete.geocoder.cit.api.here.com/6.2/suggest.json',
```
Comment (dverbovyi) [inline src/providers/api.service.js]: @aanev please move an url to CONFIG constant above of this file here: https://github.com/dverbovyi/angular-heremaps/blob/master/src/providers/api.service.js#L19

File: src/providers/api.service.js [hunk 2]
```diff
@@ -123,6 +125,66 @@ function HereMapsAPIService($q, $http, HereMapsConfig, HereMapsUtilsService, Her
...
+    }
+
+    /**
+     * Finds location by HERE Maps Location identifier. Returns XML string.
+     */
+    function findLocationById(locationId) {
+        if (!locationId)
+            return console.error('Missing Location Identifier');
+
+        var locationUrl = protocol + '://geocoder.cit.api.here.com/6.2/geocode.xml',
```
Comment (dverbovyi) [inline src/providers/api.service.js]: @aanev please move an url to CONFIG constant above of this file here: https://github.com/dverbovyi/angular-heremaps/blob/master/src/providers/api.service.js#L19

**Review by dverbovyi — CHANGES_REQUESTED** [review]

**Comment by aanev** [comment]
@dverbovyi I was thinking that these changes should not be included in HereMapsDirective. Rather the user will be able to choose to use them as a service in order to provide text-box auto-completion. And then the results can be visualized in the HereMapsDirective.

**Review by dverbovyi — APPROVED** [review]

**Comment by dverbovyi** [comment]
@aanev your PR was successfully released https://github.com/dverbovyi/angular-heremaps/releases/tag/v0.1.6

and published to npm https://www.npmjs.com/package/angular-heremaps

Thanks for contribution

**Pull request merged**